flytekit/contrib/notebook.py

```diff
@@ -0,0 +1,558 @@
+            literals.
+        :rtype: dict[Text, T]
+        :returns: The output produced by this task in Python standard format.
+        """
+        return _engine_loader.get_engine('local').get_task(self).execute(
+            _type_helpers.pack_python_std_map_to_literal_map(input_map, {
+                k: _type_helpers.get_sdk_type_from_literal_type(v.type)
+                for k, v in _six.iteritems(self.interface.inputs)
+            })
+        )
+
+    @_exception_scopes.system_entry_point
+    def execute(self, context, inputs):
+        """
+        :param flytekit.engines.common.EngineContext context:
+        :param flytekit.models.literals.LiteralMap inputs:
+        :rtype: dict[Text, flytekit.models.common.FlyteIdlEntity]
+        :returns: This function must return a dictionary mapping 'filenames' to Flyte Interface Entities.  These
+            entities will be used by the engine to pass data from node to node, populate metadata, etc. etc..  Each
+            engine will have different behavior.  For instance, the Flyte engine will upload the entities to a remote
+            working directory (with the names provided), which will in turn allow Flyte Propeller to push along the
+            workflow.  Where as local engine will merely feed the outputs directly into the next node.
+        """
+        inputs_dict = _type_helpers.unpack_literal_map_to_sdk_python_std(inputs, {
+            k: _type_helpers.get_sdk_type_from_literal_type(v.type) for k, v in _six.iteritems(self.interface.inputs)
+        })
+
+        input_notebook_path = self._notebook_path
+        # Execute Notebook via Papermill.
+        output_notebook_path = input_notebook_path + '.out'
+        _pm.execute_notebook(
+            input_notebook_path,
+            output_notebook_path,
+            parameters=inputs_dict
+        )
+
+        # Parse Outputs from Notebook.
+        outputs = None
+        with open(output_notebook_path) as json_file:
+            data = _json.load(json_file)
+            for p in data['cells']:
+                meta = p['metadata']
+                if "outputs" in meta["tags"]:
+                    outputs = ' '.join(p['outputs'][0]['data']['text/plain'])
+
+        if outputs is not None:
+            dict = _literal_models._literals_pb2.LiteralMap()
+            _text_format.Parse(outputs, dict)
+
+        # Add output_notebook as an output to the task.
+        output_notebook = _task_output.OutputReference(
+            _type_helpers.get_sdk_type_from_literal_type(_Types.Blob.to_flyte_literal_type()))
+        output_notebook.set(output_notebook_path)
+
+        output_literal_map = _literal_models.LiteralMap.from_flyte_idl(dict)
+        output_literal_map.literals[OUTPUT_NOTEBOOK] = output_notebook.sdk_value
+
+        return {
+            _constants.OUTPUT_FILE_NAME: output_literal_map
+        }
+
+    @property
+    def container(self):
```

Comment:
Can you explain why these need to be overridden from the base class?  I'm not looking too closely, but it looks pretty similar.

Reply:
The base class is SDKTask and doesn't seem to have these methods ?